filesystem/ImagickBackend.php

```diff
@@ -183,17 +183,42 @@ public function resizeByHeight($height) {
 	 *
 	 * @param int $width
 	 * @param int $height
+	 * @param int $transparencyPercent
 	 * @return Image_Backend
 	 */
-	public function paddedResize($width, $height, $backgroundColor = "FFFFFF") {
+	public function paddedResize($width, $height, $backgroundColor = "FFFFFF", $transparencyPercent=0) {
 		$new = $this->resizeRatio($width, $height);
-		$new->setImageBackgroundColor("#".$backgroundColor);
+		if($transparencyPercent) {
+			$alphaHex = $this->calculateAlphaHex($transparencyPercent);
+			$new->setImageBackgroundColor("#{$backgroundColor}{$alphaHex}");
+		} else {
+			$new->setImageBackgroundColor("#{$backgroundColor}");
+		}
 		$w = $new->getImageWidth();
 		$h = $new->getImageHeight();
 		$new->extentImage($width,$height,($w-$width)/2,($h-$height)/2);
 		
 		return $new;
 	}
+
+	/**
+	 * Convert a percentage (or 'true') to a two char hex code to signifiy the level of an alpha channel
+	 *
+	 * @param $percent
+	 * @return string
+	 */
+	public function calculateAlphaHex($percent) {
+		if($percent > 100 || $percent == 'true') {
```

Comment:
It means in the template you can then say 

    PaddedResize(100,100,FFFFFF,true)

and then it will be transparent, just like doing 100. It's a string check for true since the template parser gives it as a string and not boolean. Happy to remove thou, just thought was a nice addition :)